### PR TITLE
livecheck: enable Sorbet type checking

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?
- [ ] Have you successfully run `brew man` locally and committed any changes?

-----

### CODE REVIEW

**Review by MikeMcQuaid — APPROVED** [review]

**Review by samford — CHANGES_REQUESTED** [review]
I didn't know that Sorbet method signatures were also used in our documentation, so that's good to know. We can also continue to use `@param` to add a description for a parameter when it's beneficial, as the type information from `@param` is overridden by the Sorbet method signature (i.e., only the description is used from `@param`).

I've added one suggestion for some code we can potentially refactor but this looks good to me otherwise.

File: Library/Homebrew/livecheck/livecheck.rb [hunk 1]
```diff
@@ -268,58 +284,72 @@ def status_hash(formula_or_cask, status_str, messages = nil, full_name: false, v
...
+      if cask.present? && cask.version.latest? && !cask.livecheckable?
         return status_hash(cask, "latest", full_name: full_name, verbose: verbose) if json
 
-        puts "#{Tty.red}#{cask_name(cask, full_name: full_name)}#{Tty.reset} : latest" unless quiet
+        puts "#{Tty.red}#{name}#{Tty.reset} : latest" unless quiet
         return
       end
 
-      if cask&.url&.unversioned? && !cask.livecheckable?
+      if cask.present? && cask.url.unversioned? && !cask.livecheckable?
```
Comment (SeekingMeaning) [inline Library/Homebrew/livecheck/livecheck.rb:341]: https://github.com/sorbet/sorbet/issues/957

Comment (SeekingMeaning) [inline Library/Homebrew/livecheck/livecheck.rb:341]: https://github.com/sorbet/sorbet/issues/957

Comment (reitermarkus) [reply Library/Homebrew/livecheck/livecheck.rb:341]: `url` can still be `nil` even if `cask` isn't.

File: Library/Homebrew/livecheck/livecheck.rb [hunk 2]
```diff
@@ -268,58 +284,72 @@ def status_hash(formula_or_cask, status_str, messages = nil, full_name: false, v
...
       if formula&.versioned_formula? && !formula.livecheckable?
         return status_hash(formula, "versioned", full_name: full_name, verbose: verbose) if json
 
-        puts "#{Tty.red}#{formula_name(formula, full_name: full_name)}#{Tty.reset} : versioned" unless quiet
+        puts "#{Tty.red}#{name}#{Tty.reset} : versioned" unless quiet
         return
       end
 
-      if cask&.version&.latest? && !cask.livecheckable?
+      if cask.present? && cask.version.latest? && !cask.livecheckable?
```
Comment (reitermarkus) [inline Library/Homebrew/livecheck/livecheck.rb:334]: `version` can still be `nil` even if `cask` isn't.

File: Library/Homebrew/livecheck/livecheck.rb [hunk 3]
```diff
@@ -268,58 +284,72 @@ def status_hash(formula_or_cask, status_str, messages = nil, full_name: false, v
...
+      if cask.present? && cask.version&.latest? && !cask.livecheckable?
         return status_hash(cask, "latest", full_name: full_name, verbose: verbose) if json
 
-        puts "#{Tty.red}#{cask_name(cask, full_name: full_name)}#{Tty.reset} : latest" unless quiet
+        puts "#{Tty.red}#{name}#{Tty.reset} : latest" unless quiet
         return
       end
 
-      if cask&.url&.unversioned? && !cask.livecheckable?
+      if cask.present? && cask.url&.unversioned? && !cask.livecheckable?
```
Comment (SeekingMeaning) [inline Library/Homebrew/livecheck/livecheck.rb:341]: https://github.com/sorbet/sorbet/issues/957

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @SeekingMeaning!

**Pull request merged**